src/main/java/com/sergtm/service/impl/WeatherServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package com.sergtm.service.impl;
+
+import com.sergtm.component.WeatherDataPuller;
+import com.sergtm.dao.IWeatherDao;
+import com.sergtm.entities.Weather;
+import com.sergtm.model.ServiceName;
+import com.sergtm.model.weatherModel.WeatherModel;
+import com.sergtm.service.IStatusService;
+import com.sergtm.service.IWeatherService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+
+@Service
+public class WeatherServiceImpl implements IWeatherService {
+    private static final String OPEN_WEATHER_MAP_URL = "http://openweathermap.org/img/w/%s.png";
+
+    @Autowired
+    private WeatherDataPuller weatherDataPuller;
+
+    @Autowired
+    private IStatusService statusService;
+
+    @Autowired
+    private IWeatherDao weatherDao;
+
+    @Transactional
+    @Override
+    public Weather getWeather() {
+        if (statusService.identifyLastModifiedService().equals("None")) {
```

Comment:
Compare with enum value. If need change return value of the above function to enum value